apps/api/src/modules/job/job.repository.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@nestjs/common'
+import { Casting, Job, Prisma } from '@prisma/client'
+import { PrismaService } from 'src/database/prisma.service'
+
+import { GetJobCardDto } from './job.dto'
+
+@Injectable()
+export class JobRepository {
+  constructor(private prisma: PrismaService) {}
+
+  async getJobCount(): Promise<number> {
+    return await this.prisma.job.count()
+  }
+
+  async getJobJoined(params: {
+    skip?: number
+    take?: number
+    cursor?: Prisma.JobWhereUniqueInput
+    where?: Prisma.JobWhereInput
+    orderBy?: Prisma.JobOrderByWithRelationInput
+  }): Promise<GetJobCardDto[]> {
+    //generate join casting and user table params
+    const paramsWithInclude = {
+      ...params,
+      include: {
+        Casting: {
+          include: {
+            User: true,
+          },
+        },
+      },
+    }
+    //get data
+    const jobs = await this.prisma.job.findMany(paramsWithInclude)
+    //select fields of GetJobDto
+    const selectedFields = jobs.map((job) => ({
+      jobId: job.jobId,
+      title: job.title,
+      companyName: job.Casting.companyName,
+      description: job.description,
+      status: job.status,
+      actorCount: job.actorCount,
+      gender: job.gender,
+      wage: job.wage,
+      applicationDeadline: job.applicationDeadline,
+      jobCastingImageUrl: job.Casting.User.profileImageUrl,
+    }))
```

Comment:
Not sure that this is the cleaner way to write this or not (but it will cause unused var that we must make eslint ignore)
@palmcm what do you think ?
```ts
    const selectedFields = jobs.map(
      ({ Casting, minAge, maxAge, updatedAt, createdAt, ...rest }) => ({
        ...rest,
        jobCastingImageUrl: Casting.User.profileImageUrl,
        companyName: Casting.companyName,
      }),
    )
 ```

Reply:
At first, I was picking the field one by one according to API design.
The way you suggested would be cleaner. But in the future, if the Job schema is updated with a new field the API shall return them too?
So, Should I change?

Reply:
It might be hard to edit if the schema is change, but if you trust in PKhing you can do it (jk). It's just cleaner